extensions/wikia/UserProfilePageV3/maintenance/migrateAvatarsViaService.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/**
+ * Script migrates the user avatars from DFS to user avatars service
+ *
+ * @see PLATFORM-1419
+ *
+ * @author Macbre
+ * @ingroup Maintenance
+ */
+
+putenv( 'SERVER_ID=177' ); // run in the context of a community wiki
+
+require_once( __DIR__ . '/../../../../maintenance/Maintenance.php' );
+
+class AvatarsMigratorException extends Exception {}
+
+/**
+ * Maintenance script class
+ */
+class AvatarsMigrator extends Maintenance {
+
+	private $isDryRun = true;
+
+	/**
+	 * Set script options
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		$this->addOption('dry-run', 'Don\'t perform any operations [default]');
+		$this->addOption( 'force', 'Apply the changes made by the script' );
+
+		$this->mDescription = 'This script migrates the user avatars from DFS to user avatars service';
+	}
+
+	public function execute() {
+		// read options
+		$this->isDryRun = $this->hasOption( 'dry-run' ) || !$this->hasOption( 'force' );
+
+		if ($this->isDryRun) $this->output( "Running in dry-run mode!\n\n" );
+
+		$this->output( "Getting the list of all accounts...\n" );
+
+		// get all accounts
+		global $wgExternalSharedDB;
+		$db = $this->getDB( DB_SLAVE, [], $wgExternalSharedDB );
+
+		$res = $db->select(
+			'`user`',
+			'user_id AS id',
+			[],
+			__METHOD__
+		);
```

Comment:
@jsutterfield, Is the user preferences service going to keep storing preferences in `user_preferences` table or do we need to get the list of all accounts and access preferences via `User` object?

In other words: can we perform the following query on `wikicities` to get the list of all accounts that require avatars migration:

```sql
select * from user_properties where up_property = 'avatar';
```